lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -125,15 +126,15 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
         """
         try:
             index_map = Metadata.getvalue(dataset=dataset, key=Metadata.INDEX_MAP)
-        except MetadataError as e:
-            self.logger.error(f"Indices from metadata table not found {e!r}")
-            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
-
-        if index_map is None:
+        except MetadataError:
             self.logger.error(
                 f"server.index-map not found in Metadata for a dataset {dataset!r}"
             )
-            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+            raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR)
```

Comment:
We're not logging the exception, here...is there no information in the exception object or the traceback that we want to collect?  (And, in the message that we _are_ logging, we're **assuming** that the failure is for `server.index-map`, rather than using the actual input key (`Metadata.INDEX_MAP`) or the reported failure value (`MetadataError.key`)...which, presumably, _are_ `server.index-map`, but there's no guarantee that that will always be the case.)

Reply:
I wouldn't worry about the fact that we're doing a `logger.error` here, since the original `MetadataError` will be implicitly chained to the `APIAbort`, and `_dispatch` *does* `logger.exception`. I suppose we could almost argue that the `logger.error` here is redundant, although it does mention the dataset name which might not otherwise be clear.

In principle, yeah, it would have been cleaner to format `Metadata.INDEX_MAP` as a parameter rather than as a literal. I suppose that's my fault, along with the original bad language of the messages. 😄 

Do we need to improve this now?

Reply:
If `_dispatch` is going to log the entire exception chain, then clearly we don't need to log the `MetadataError` exception traceback here.  😅

My principal concern is that the stringized `MetadataError` exception object reports the dataset and key names (and sometimes other stuff), and I think we've lost that information with this change.  (The traceback doesn't include the exception object string, does it?).

So, I think we should either log the `MetadataError` exception string (e.g., instead of the current log message at line 131), or we should use that string in the `APIAbort` (and omit the `logger.error()` call here).